### ISSUE TITLE
Default PBR shader support

### ISSUE THREAD

**Issue by vpenades** [mellinoe/veldrid#169]
Nowadays, PBR shader material is basically a de-facto standard.

Given that almost anybody using Veldrid will have to use PBR, maybe it could be worth to implement a default PBR shader (hopefully with Skinning support).

I agree it should not go into the base Veldrid package... but having predefined shaders like these will really help!

**Comment by mellinoe** [comment]
Yes, I agree a PBR example is very valuable, and I'd like to add one at some point. I think it's less about the shaders themselves (you can just find those online and use them with Veldrid), but more about how to set up the Pipelines, ResourceSets, etc. in Veldrid to feed the data in properly.

On that front, I am hoping to start building up a better designed and more accessible "sample gallery" in this repo here, as a long-term replacement for the veldrid-samples repo. Since 4.6.0 is out now, that sample gallery is next on my list of things to do, but I don't really have a timeline for when it will be usable, or when something like a PBR example could be added to it.

**Comment by gleblebedev** [comment]
I'm working towards Veldrid-based GLTF viewer. I have some progress:
https://gyazo.com/81b59aa949681832bea8954837d142ab

The PBR shader is basically translation of a PBR shader from Kronos sample. @vpenades, would it be useful for your purpose? The original shader has skinning and blend shapes support although I didn't get to it yet.

**Comment by vpenades** [comment]
@gleblebedev Sure, actually I'm working on a glTF renderer for MonoGame too! we're planning to support Veldrid as soon as possible, but we're still maintaining some legacy code under MonoGame.

On a side note, look forward for SharpGLTF.Runtime namespace, it's a helper to facilitate rendering glTF files on any c# graphics engine, I hope it will be very helpful to allow rendering glTF models with animations.

**Comment by gleblebedev** [comment]
I'm using the SharpGLTF for the viewer app:
https://github.com/gleblebedev/VeldridGlTF

**Comment by RUSshy** [comment]
@vpenades is the renderer available somewhere, i'm curious to see how it's done using monogame

**Comment by vpenades** [comment]
@RUSshy no, the monogame renderer is something I'm pulling out of neccessity for an internal project.

But the relevant part, the code that governates how things have to be rendered, including skinning and morphing, will be available soon in SharpGLTF, so any rendering engine could benefit from it.
